composer/datasets/mnist_hparams.py

```diff
@@ -52,12 +52,21 @@ def initialize_object(self, batch_size: int, dataloader_hparams: DataLoaderHpara
                 raise ValueError('datadir is required if synthetic is False')
 
             transform = transforms.Compose([transforms.ToTensor()])
-            dataset = datasets.MNIST(
-                self.datadir,
-                train=self.is_train,
-                download=self.download,
-                transform=transform,
-            )
+
+            def _get_dataset(download: bool):
```

Comment:
Nit: make this a lambda? E.g.
`_get_dataset = lambda download: datasets.MNIST(...)`